src/test/cluster_health.zig

```diff
@@ -0,0 +1,117 @@
+const std = @import("std");
+const assert = std.debug.assert;
+const mem = std.mem;
+const log = std.log.scoped(.health);
+
+pub const ClusterHealthOptions = struct {
+    replica_count: u8,
+    /// The minimum number of replicas that must be up.
+    //replica_count_min: u8,
+    seed: u64,
+
+    /// Probability per tick that a crash will occur.
+    crash_probability: f64,
+    /// Minimum duration of a crash.
+    crash_stability: u32,
+    /// Probability per tick that a crashed replica will recovery.
+    restart_probability: f64,
+    /// Minimum time a replica is up until it is crashed again.
+    restart_stability: u32,
+};
+
+const ReplicaHealth = union(enum) {
+    /// When >0, the replica cannot crash.
+    /// When =0, the replica may crash.
+    up: u32,
+    /// When >0, this is the ticks remaining until recovery is possible.
+    /// When =0, the replica may recover.
+    down: u32,
+};
+
+pub const ClusterHealth = struct {
+    options: ClusterHealthOptions,
+    prng: std.rand.DefaultPrng,
```

Comment:
```suggestion
    random: std.rand.Random,
```

It's the same convention as `std.mem.Allocator`, so that we can then swap out PRNGs without being restricted to the default PRNG.